presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -408,6 +423,10 @@ protected Boolean visitDereferenceExpression(DereferenceExpression node, Void co
 
         private Boolean isField(QualifiedName qualifiedName)
         {
+            if (lambdaVariableNames.contains(qualifiedName.toString())) {
```

Comment:
I think it should be enough to check if `Identifier` is not a column reference (lambda variable references aren't). This way `AggregationAnalyzer` code would simplify and you wouldn't have to use and maintain `lambdaVariableNames` map.

If column reference approach is not strong enough, then I think you could keep in `Analysis` a set:
```
Set<Expression> lambdaReference
```
which would explicitly specify which expressions are lambda variable reference.